### PR TITLE
[NCL-4396] User, SCMRepository, Environment REST endpoints

### PR DESCRIPTION
**Pull request by thescouser89**
Missing SCMRepository creation: will implement later

### Checklist:

* [ ] Have you added a note in the CHANGELOG.md for your change if user-facing?
* [ ] Have you added unit tests for your change?

### CODE REVIEW

File: rest-new/src/main/java/org/jboss/pnc/rest/endpoints/UserEndpointImpl.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+            // if user already in the database
+            if (currentUser != null) {
+                return currentUser;
+            }
+
+            // else add user in database
+            currentUser = User.builder().username(loggedUser).build();
+            return super.create(currentUser);
+        }
```
Comment (janinko) [inline rest-new/src/main/java/org/jboss/pnc/rest/endpoints/UserEndpointImpl.java]: I don't like this
 - String interning
 - Synchronization 
 - Bussiness logic in view

Do we really need the synchronization? Aren't transactions enough? I guess some just some get-or-create method from provider shoud be called here.

**Comment by thescouser89** [comment]
retest this please

**Comment by thescouser89** [comment]
retest this please

**Comment by thescouser89** [comment]
retest this please

File: facade/src/main/java/org/jboss/pnc/facade/providers/BuildProviderImpl.java [hunk 2]
```diff
@@ -46,14 +47,8 @@
     private BuildConfigurationRepository buildConfigurationRepository;
 
     @Inject
-    public BuildProviderImpl(BuildRecordRepository repository,
-                             BuildMapper mapper,
-                             EntityManager entityManager,
-                             BuildConfigurationRepository buildConfigurationRepository) {
-
+    public BuildProviderImpl(BuildRecordRepository repository, BuildMapper mapper) {
```
Comment (janinko) [inline facade/src/main/java/org/jboss/pnc/facade/providers/BuildProviderImpl.java]: The entityManager and buildConfigurationRepository are nto needed anymore? If not delet the fields too.

Comment (thescouser89) [reply facade/src/main/java/org/jboss/pnc/facade/providers/BuildProviderImpl.java]: BuildConfigurationRepository still needed. Sorry for that error!

**Pull request merged**